tests/planner.spec.ts

```diff
@@ -0,0 +1,142 @@
+import { test, expect, Page } from "@playwright/test";
+
+const testForArcanistPlanning = (
+  ...arcanists: {
+    name: string
+    insight: [number, number]
+    level: [number, number]
+    resonance: [number, number]
+    expectMaterials: {
+      name: string
+      count: string
+    }[]
+  }[]
+) => {
+  return async ({ page }: { page: Page }) => {
+    const setupStatus = async (blockName: string, status: [string, string]) => {
+      if (status.length !== 2) { return }
+
+      const settingBlock = page.locator('.edit-overlay')
+        .locator(`:below(:text("${blockName}"))`)
+        .first()
+
+      for (let i = 0; i < status.length; i++) {
+        await settingBlock.getByRole('button').nth(i).click()
+        const fromBlock = settingBlock.locator('div:below(button)')
+          .filter({ has: page.getByRole('button') })
+          .first()
+        await fromBlock.getByRole('button', { name: status[i], exact: true }).click()
+      }
+    }
+
+    for (let i = 0; i < arcanists.length; i++) {
+      const arcanist = arcanists[i]
+      await page.getByRole("button", { name: "Add Arcanist" }).click()
+      await page.getByLabel('Show Unreleased Arcanists').check()
+      await page.getByText(arcanist.name).click()
+
+      await setupStatus('CURRENT LEVEL', [
+        arcanist.insight[0] ? `Option ${arcanist.insight[0]}` : '', `${arcanist.level[0]}`
```

Comment:
This code has two issues:
1. You used an implicit type conversion to check for insights with a `0` check; better to define a boolean variable to make this readable.
2. Finding the icon button by looking for a '' named button is hard to understand (yeh...i know we did the same things in `tracker.spec.ts`). We should find another way to locate the option button.

Reply:
2. is it better to add some test-id props to our project?

Reply:
yeh, that's good idea.